### PR TITLE
Fix route shown even if input is altered

### LINKED ISSUE #144: Route shown even if waypoint input is altered
### What is the problem you are facing
Right now when we change the input of any waypoint, the route is still shown.

### What is your suggested solution
Any change made to input should remove the route automatically just as what happens when we click the cross icon to remove the waypoint manually.

### Screenshots

https://user-images.githubusercontent.com/55504616/227126913-7ffca138-cc71-45cc-a84b-65fffc6f6dd3.MOV

### PR DESCRIPTION
**Pull request by Sheikh-JamirAlam**
## 🛠️ Fixes Issue
Fixes #144 

## 👨‍💻 Changes proposed
Now we can alter the waypoint input field and the route will disappear.

## 📄 Note to reviewers
Please see the attached video.

## 📷 Screenshots

https://user-images.githubusercontent.com/48979376/227571796-9e8fa0de-408b-48f4-95f9-5c68375e7abe.mp4

### CODE REVIEW

**Review by Ananya2001-an — COMMENTED** [review]
Hey @Sheikh-JamirAlam Thanks for the quick PR. But actually this is not going to work, every time we make changes to the input it will clear the routes which means nothing is getting shown. Did you test it properly? What we want is that when someone starts altering a waypoint's input value then that marker as well as the route associated to that point should get removed. I hope you get it now. Feel free to ask if you still have any doubt :)

**Comment by Sheikh-JamirAlam** [comment]
@Ananya2001-an Sorry I think I don't get it. What do you mean by altering a waypoint's input value?

**Comment by Ananya2001-an** [comment]
> @Ananya2001-an Sorry I think I don't get it. What do you mean by altering a waypoint's input value?

In the screenshot in issue #144 I showed that once we start changing the input value of a waypoint the marker disappears but the route is still there. So what we want is to remove it if after drawing a route someone starts changing the input (waypoint). But your code is not letting routes getting drawn on the map at all. It will just clear up every time someone types into the search box. Hope u get it now. Please play with the code a bit to understand which code is affecting which part of the app.

**Comment by Sheikh-JamirAlam** [comment]
@Ananya2001-an Sorry for the late response, I think I have managed to fix it. Am sharing the video and commint my changed now. Also I created a function and named it `isWaypoint`, hope the naming is fine.

https://user-images.githubusercontent.com/48979376/227718462-08836a77-84ab-4e46-b1ac-0df51fc369bd.mp4

File: src/actions/directionsActions.js [hunk 1]
```diff
@@ -311,6 +311,13 @@ export const doRemoveWaypoint = (index) => (dispatch, getState) => {
   dispatch(updatePermalink())
 }
 
+export const isWaypoint = (index) => (dispatch, getState) => {
+  const waypoints = getState().directions.waypoints
+  if (waypoints[index].geocodeResults.length > 0) {
+    dispatch(clearRoutes(VALHALLA_OSM_URL))
```
Comment (Ananya2001-an) [inline src/actions/directionsActions.js:317]: This will clear all the routes, we just want to remove the one that was associated with that marker.

**Comment by Ananya2001-an** [comment]
> @Ananya2001-an Sorry for the late response, I think I have managed to fix it. Am sharing the video and commint my changed now. Also I created a function and named it `isWaypoint`, hope the naming is fine.

Don't stress about it. It's not a simple issue. Try to work according to your pace. You are already doing good 👍🏼

File: src/actions/directionsActions.js (hunk 1 above)
Comment (nilsnolde) [reply src/actions/directionsActions.js:317]: I think that's fine no? Are you thinking to just remove a "section" of the route? So if there's 3 waypoints and a user alters the very last one, you'd like to just remove the route's section between the 2nd and last waypoint? 

That'd be possible but IMHO unnecessary as we'd send a new route request with _all_ waypoints anyways (just to keep things simple).

Comment (Ananya2001-an) [reply src/actions/directionsActions.js:317]: Yeah, I guess you are right. Maybe we should keep it simple. I think this is good enough. I was making it complex in terms of UX but I guess it's not important unless it turns out to be a concern in the future.

Comment (Sheikh-JamirAlam) [reply src/actions/directionsActions.js:317]: @nilsnolde @Ananya2001-an So, will this get merged?

File: src/actions/directionsActions.js [hunk 2]
```diff
@@ -364,6 +371,7 @@ export const doAddWaypoint = (doInsert) => (dispatch, getState) => {
     isFetching: false,
     userInput: '',
   }
+  //console.log(emptyWp)
```
Comment (nilsnolde) [inline src/actions/directionsActions.js:374]: Pls remove comments

**Review by nilsnolde — COMMENTED** [review]
Got my approval, just remove the comments. Thanks again!

**Review by Ananya2001-an — APPROVED** [review]
Looks good! 👍🏼 Just remove the comment as Nils mentioned above.

File: src/actions/directionsActions.js (hunk 2 above)
Comment (Sheikh-JamirAlam) [reply src/actions/directionsActions.js]: Done

**Pull request merged**